lib/core/managementClient.ts

```diff
@@ -141,29 +174,61 @@ export class ManagementClient extends LinkEntity {
    * @returns Promise<ReceivedSBMessage[]>
    */
   async peek(messageCount?: number): Promise<ReceivedMessageInfo[]> {
-    return this.peekBySequenceNumber(this._lastPeekedSequenceNumber.add(1), messageCount);
+    return this.peekBySequenceNumber(this._lastPeekedSequenceNumber.add(1), {
+      messageCount: messageCount
+    });
+  }
+
+  /**
+   * Fetches the next batch of active messages in the current MessageSession. The first call to
+   * `peek()` fetches the first active message for this client. Each subsequent call fetches the
+   * subsequent message in the entity.
+   *
+   * Unlike a `received` message, `peeked` message will not have lock token associated with it,
+   * and hence it cannot be `Completed/Abandoned/Deferred/Deadlettered/Renewed`. Also, unlike
+   * `receive() | receiveBatch()` this method will also fetch `Deferred` messages, but
+   * **NOT** `Deadlettered` messages.
+   * @param {string} sessionId The sessionId from which messages need to be peeked.
+   * @param {number} [messageCount] The number of messages to retrieve. Default value `1`.
+   * @returns Promise<ReceivedMessageInfo[]>
+   */
+  async peekMessagesBySession(sessionId: string, messageCount?: number): Promise<ReceivedMessageInfo[]> {
+    if (sessionId == undefined) {
+      throw new Error("'sessionId' is a required parameter and must be of type 'string'.");
+    }
+    return this.peekBySequenceNumber(this._lastPeekedSequenceNumber.add(1), {
+      sessionId: sessionId,
+      messageCount: messageCount
+    });
   }
 
   /**
    * Peeks the desired number of messages from the specified sequence number.
    * @param {Long} fromSequenceNumber The sequence number from where to read the message.
-   * @param {number} [messageCount] The number of messages to retrieve. Default value `1`.
-   * @returns Promise<ReceivedSBMessage[]>
+   * @param {PeekOptions} [options] Options that can be provided while peeking messages.
```

Comment:
The sessionId is an optional parameter for receiveDeferred* method. That is the case with peekBySequenceNumber() as well. However, peekBySequenceNumber has more optional parameters, hence the optional options object, whereas receiveDefered* only has one optional parameter. Creating an options object over there will be an overkill. I don't think there will be any more optional parameters in the near future for receiveDefered* methods

Reply:
Since neither of these functions are exposed to the user directly, I believe my earlier concern for consistency from the user's perspective is invalid.